test/test_um2netcdf.py

```diff
@@ -651,6 +663,355 @@ def test_fix_units_do_nothing_no_um_units(ua_plev_cube):
+    # This grid is offset half a grid cell compared to the standard
+    # New Dynamics longitude grid.
+    lon_u_points = np.arange(0, 360, D_LON_N96, dtype="float32")
+    return lon_u_points, um2nc.LONGITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lat_standard_nd_coord():
+    # iris DimCoord imitating the standard latitude
+    # grid from ESM1.5 (which uses the New Dynamics grid).
+    lat_points = np.arange(-90, 91, D_LAT_N96, dtype="float32")
+    return lat_points, um2nc.LATITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lon_standard_nd_coord():
+    # iris DimCoord imitating the standard longitude
+    # grid from ESM1.5 (which uses the New Dynamics grid).
+    lon_points = np.arange(0, 360, D_LON_N96, dtype="float32")
+    return lon_points, um2nc.LONGITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lat_standard_eg_coord():
+    # iris DimCoord imitating the standard latitude
+    # grid from CM2 (which uses the End Game grid).
+    lat_points = np.arange(-90 + 0.5*D_LAT_N96, 90.,
+                           D_LAT_N96, dtype="float32")
+    return lat_points, um2nc.LATITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lon_standard_eg_coord():
+    # iris DimCoord imitating the standard longitude
+    # grid from CM2 (which uses the End Game grid).
+    lon_points = np.arange(0.5*D_LON_N96, 360, D_LON_N96, dtype="float32")
+    return lon_points, um2nc.LONGITUDE
+
+
+class DummyCubeWithCoords(DummyCube):
+    # DummyCube with coordinates, which can be filled with
+    # iris.DimCoord objects for testing.
+    def __init__(self, dummy_cube, coords=None):
+        super().__init__(dummy_cube.item_code,
+                         dummy_cube.var_name,
+                         dummy_cube.attributes,
+                         dummy_cube.units)
+
+        # Set up coordinate dictionary to hold each dummy coordinate, with keys
+        # given by the coordinate names. This ensures that the key used to
+        # access a coordinate via the coord method matches the
+        # coordinate's name.
+        if coords is not None:
+            self.coordinate_dict = {
+                coord.name(): coord for coord in coords
+            }
+        else:
+            self.coordinate_dict = {}
+
+    def coord(self, coordinate_name):
+        return self.coordinate_dict[coordinate_name]
+
+
+def assert_coordinates_are_unmodified(lat_coord, lon_coord):
+    """
+    Helper function to check that a coordinate's attributes match
+    those expected for a coordinate that has not yet been modified
+    by fix_latlon_coords.
+    """
+    for coord in [lat_coord, lon_coord]:
+        assert coord.points.dtype == np.dtype("float32")
+        assert not coord.has_bounds()
+        assert coord.var_name is None
+
+
+def assert_dtype_float64(lat_coord, lon_coord):
+    assert lat_coord.points.dtype == np.dtype("float64")
+    assert lon_coord.points.dtype == np.dtype("float64")
+
+
+def assert_has_bounds(lat_coord, lon_coord):
```

Comment:
This could also be switched to `has_bounds()` as per the stylistic fix for the float type check.

Reply:
Do you think it's clearer to do this, e.g:
```python
def has_bounds(lat_coord, lon_coord):
    return lat_coord.has_bounds and lon_coord.has_bounds()

...
def test_some_function():
...
assert has_bounds(lat_coord, lon_coord)
```
or to instead just go back to what we had previously and directly use the coordinates' `has_bounds` method in the tests:

```python
def test_some_function():
...
assert lat_coord.has_bounds()
assert lon_coord.has_bounds()
```
Let me know what you prefer!

Reply:
The first option provides a degree of flexibility. Each test just has to assert the helper func. If the bool check changes, it can be fixed in 1 place & then applies to all dependent tests.

This is one of those fiddlier test cases where the iterations are less clear due to existing code, lib dependencies etc.

Reply:
Sounds good! Added in 6a4c876